src/player.cpp

```diff
@@ -7456,6 +7456,59 @@ void player::rooted()
     }
 }
 
+bool player::add_faction_warning( const faction_id id )
+{
+    std::map<faction_id, std::pair<int, time_point>>::iterator it;
+    it = warning_record.find( id );
+    if( it != warning_record.end() ) {
+        warning_record.find( id )->second.first += 1;
+        if( warning_record.find( id )->second.second - calendar::turn > 5_minutes ) {
+            warning_record.find( id )->second.first -= 1;
+        }
+        warning_record.find( id )->second.second = calendar::turn;
+        if( warning_record.find( id )->second.first > 3 ) {
+            return true;
+        }
+    } else {
+        warning_record[id] = std::make_pair( 1, calendar::turn );
+    }
+    faction *fac = g->faction_manager_ptr->get( id );
+    if( fac != nullptr && this->is_player() ) {
+        fac->likes_u -= 1;
+        fac->respects_u -= 1;
+    }
+    add_msg( "current warnings: %d", current_warnings_fac( id ) );
```

Comment:
Maybe you meant it to be debug message?
```suggestion
    add_msg( m_debug, "current warnings: %d", current_warnings_fac( id ) );
```

Reply:
Woops no, meant to delete it.